Angular/angular-mentoring/src/app/pipes/filter.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filter'
+})
+export class FilterPipe implements PipeTransform {
+
+  transform(values: any[], filter: string): any {
+    if (filter != undefined) {
+      filter = filter.toLowerCase();
+      values = values.filter(value => {
+        const { title, content } = value;
+        return title.toLowerCase().includes(filter) || content.toLowerCase().includes(filter);
+      });
+    }
+
+    return values;
```

Comment:
We could make this code easier to read, by returning earlier and avoiding nesting.

This is simple code, but in a more complex situation, nesting makes it more difficult to follow the code and understanding what's happening.

Consider below:

```typescript
const emptyFilter = (filter ?? '').length === 0;

if (emptyFilter) {
  return values;
}

filter = filter.toLowerCase();
values = values.filter({ title, content } => {
  return title.toLowerCase().includes(filter) || content.toLowerCase().includes(filter);
});

return values;
```